test/docker/http-notification-webhook/main.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+
+	"kpt.dev/configsync/e2e/nomostest"
+)
+
+// mux is used to synchronize operations across the various endpoints
+var mux sync.Mutex
+
+// store is a simple in memory storage of notification records.
+var store nomostest.NotificationRecords
+
+func doGet(w http.ResponseWriter, r *http.Request) {
```

Comment:
suggest renaming to `doGetAll` and `doDeleteAll`. Depending on the validation, you might need another method like `doGet` with an id.

Reply:
I agree with the general premise, however I'm not sure that we have
- a use case for querying individual notifications
- a unique identifier for querying an individual notification

We could use something like an auto-incrementing integer or UUID as an ID, however I don't see that as being very usable in this use case. WDYT?

Reply:
Yes, that makes sense. We can change it later when needed.